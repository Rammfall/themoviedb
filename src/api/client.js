import axios from 'axios'
import Cookies from 'js-cookie'

const baseURL = 'http://localhost:8080/api/'

const client = axios.create({
  baseURL,
  timeout: 10000
})

client.interceptors.request.use((config) => {
  const newConfig = config

  const token = Cookies.get('auth_token')
  if (token) {
    newConfig.headers = { Authorization: `Bearer ${token}` }
  }

  return newConfig
})

client.interceptors.response.use(
  response => response,
  (error) => {
    if (error.response) {
      const { message } = error.response.data
      const { status } = error.response

      const newError = new Error(message)
      newError.status = status

      return Promise.reject(newError)
    }

    return Promise.reject(error)
  }
)

export default client
