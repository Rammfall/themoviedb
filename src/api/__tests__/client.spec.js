import Cookies from 'js-cookie'

import client from '../client'

describe('check http client', () => {
  describe('make get request', () => {
    it('success response', async () => {
      expect(client).toMatchSnapshot()
    })

    it('check cookies interceptor', async () => {
      jest.mock('js-cookie')
      Cookies.get = jest.fn(() => '0123456789')

      jest.mock('../client', (clientMock) => ({
        ...clientMock,
        get: jest.fn()
      }))

      expect(client.get('https://test.tet')).toMatchSnapshot()
      await expect(
        client.interceptors.response.handlers[0].fulfilled('Success')
      ).toStrictEqual('Success')
      await expect(
        client.interceptors.request.handlers[0].fulfilled({})
      ).toStrictEqual({ headers: { Authorization: `Bearer 0123456789` } })
    })

    it('check error interceptor', async () => {
      await expect(
        client.interceptors.response.handlers[0].rejected({
          response: {
            statusText: 'NotFound',
            status: 404,
            data: { message: 'Page not found' }
          }
        })
      ).rejects.toThrowError('Page not found')

      await expect(
        client.interceptors.response.handlers[0].rejected({
          message: 'Some error'
        })
      ).rejects.toStrictEqual({ message: 'Some error' })
    })
  })
})
