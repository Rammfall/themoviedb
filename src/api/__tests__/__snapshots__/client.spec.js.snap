// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check http client make get request check cookies interceptor 1`] = `Promise {}`;

exports[`check http client make get request success response 1`] = `[Function]`;
